homeassistant/components/notify/html5.py

```diff
@@ -129,7 +133,87 @@ def post(self, request):
             return self.json_message('Error saving registration.',
                                      HTTP_INTERNAL_SERVER_ERROR)
 
-        return self.json_message("Push notification subscriber registered.")
+        return self.json_message('Push notification subscriber registered.')
+
+
+class HTML5PushCallbackView(HomeAssistantView):
+    """Accepts push registrations from a browser."""
+
+    requires_auth = False
+    url = '/api/notify.html5/callback'
+    name = 'api:notify.html5/callback'
+
+    def __init__(self, hass, json_path):
+        """Init HTML5PushCallbackView."""
+        super().__init__(hass)
+        self.json_path = json_path
+
+    def decode_jwt(self, token):
+        """Find the registration that signed this JWT and return it."""
+        import jwt
+        registrations = _load_config(self.json_path)
+        for reg in registrations.values():
+            auth_key = reg['subscription']['keys']['auth']
+            try:
+                payload = jwt.decode(token, auth_key, verify_exp=False)
+            except jwt.exceptions.DecodeError:
+                continue
+            # pylint: disable=broad-except
+            except Exception as jwt_decode_error:
```

Comment:
Think this can just be [`jwt.exceptions.InvalidToken`](https://github.com/jpadilla/pyjwt/blob/master/jwt/exceptions.py) ?